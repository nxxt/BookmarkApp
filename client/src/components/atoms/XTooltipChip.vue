<template>
  <v-tooltip bottom>
    <template v-slot:activator="{ on }">
      <div
        class="x-tooltip-chip px-2"
        :class="{ ...classes, [color]: true }"
        :style="{ borderRadius: '5px' }"
        v-on="on"
      >
        <slot></slot>
      </div>
    </template>

    <!-- tooltip content -->
    <span>{{ tooltip }}</span>
  </v-tooltip>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component({})
export default class XTooltipChip extends Vue {
  @Prop({ type: [String, Number], default: 'tooltip' })
  tooltip!: string | number

  @Prop({ type: Object, default: () => ({}) })
  classes!: { [key: string]: boolean }

  @Prop({ type: [String, Number], default: 'blue' })
  color!: string | number

  onDuplicateTagName() {
    this.$emit('duplicate-tag-name')
  }
}
</script>
