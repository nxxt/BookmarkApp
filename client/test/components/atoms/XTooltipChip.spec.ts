import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import XTooltipChip from '@/components/atoms/XTooltipChip.vue'

const localVue = createLocalVue()

describe('XTooltipChip.vue', () => {
  // vuetify の準備
  let vuetify: typeof Vuetify
  beforeEach(() => {
    vuetify = new Vuetify()
  })

  // propsで渡す値の準備
  const propsData = {
    tooltip: 'test',
    classes: { test: true },
    color: 'dark'
  }

  const factory = (options?: any) => {
    return mount(XTooltipChip, {
      localVue,
      vuetify,
      ...options
    })
  }

  // TEST
  it('props', () => {
    // コンポーネントラッパーの作成
    const wrapper = factory({ propsData })

    // VueインスタンスのpropsがpropsDataと等しいか
    expect(wrapper.props()).toStrictEqual(propsData)
  })

  describe('template', () => {
    it('slot', () => {
      // コンポーネントラッパーの作成
      const wrapper = factory({
        slots: {
          default: '<div data-test="slotContent">slot content</div>'
        }
      })

      const slotContent = wrapper.find('[data-test="slotContent"]')
      expect(slotContent.exists()).toBe(true)
      expect(slotContent.text()).toBe('slot content')
    })

    it('snapshot', () => {
      // コンポーネントラッパーの作成
      const wrapper = factory({ propsData })

      // レンダリング結果が前回のスナップショットと同じか
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
