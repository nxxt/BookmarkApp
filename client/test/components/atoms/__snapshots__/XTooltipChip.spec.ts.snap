// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XTooltipChip.vue template snapshot 1`] = `"<span class=\\"v-tooltip v-tooltip--bottom\\"><!----><div class=\\"x-tooltip-chip px-2 test dark\\" style=\\"border-radius: 5px;\\"></div></span>"`;
